ledger/src/bank_forks.rs

```diff
@@ -218,6 +239,15 @@ impl BankForks {
         );
     }
 
+    fn check_interval(root: Slot, parent_slot: Slot, snapshot_interval_slots: usize) -> bool {
+        let interval = snapshot_interval_slots as u64;
+        assert!(parent_slot < root || root == 0);
+        assert!(interval > 0);
+        (root - parent_slot > interval)
+            || (root % interval == 0)
+            || (root % interval <= parent_slot % interval)
```

Comment:
Thisi last check `(root % interval <= parent_slot % interval)` seems like it will vary if we are traversing banks backwards (see comment above)